src/libraries/System.IO.Packaging/tests/Tests.NetFramework.cs

```diff
@@ -0,0 +1,28 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Xunit;
+
+namespace System.IO.Packaging.Tests
+{
+    public partial class Tests : FileCleanupTestBase
+    {
+        [WindowsFullFrameworkOnlyFact]
+        public void Cannot_Create_Package_On_Unseekable_Stream()
+        {
+            var documentPath = "untitled.txt";
+            Uri partUriDocument = PackUriHelper.CreatePartUri(new Uri(documentPath, UriKind.Relative));
+
+            using (MemoryStream ms = new MemoryStream())
+            using (WrappedStream ws = new WrappedStream(ms, true, true, false))
+            {
+                Assert.Throws<ArgumentException>(() => Package.Open(ws, FileMode.Create, FileAccess.Write));
```

Comment:
I see the .NET Framework version of Package.Open goes through [here](https://referencesource.microsoft.com/#WindowsBase/Base/System/IO/Packaging/Package.cs,263) which in turns calls [this](https://referencesource.microsoft.com/#WindowsBase/Base/System/IO/Packaging/Package.cs,1050) with  `streaming` set to `false`.

But then I'm not sure who's throwing `ArgumentException`. Do you have a callstack?

Reply:
It's the `streaming` parameter which generates it. It gets passed into [ZipPackage](https://referencesource.microsoft.com/#WindowsBase/Base/System/IO/Packaging/ZipPackage.cs,437), then into [ZipArchive.OpenOnStream](https://referencesource.microsoft.com/#WindowsBase/Base/MS/Internal/IO/Zip/ZipArchive.cs,126), then [ValidateModeAccessShareStreaming](https://referencesource.microsoft.com/#WindowsBase/Base/MS/Internal/IO/Zip/ZipArchive.cs,524), then [ValidateModeAccessStreamStreamingCombinations](https://referencesource.microsoft.com/#WindowsBase/Base/MS/Internal/IO/Zip/ZipArchive.cs,569), where an exception is thrown if `streaming` and `CanSeek` are both false.

The full call stack is:
```
ZipArchive.ValidateModeAccessStreamStreamingCombinations(Stream stream, FileAccess access, Boolean streaming)
ZipArchive.ValidateModeAccessShareStreaming(Stream stream, FileMode mode, FileAccess access, FileShare share, Boolean streaming)
ZipArchive.OpenOnStream(Stream stream, FileMode mode, FileAccess access, Boolean streaming)
ZipPackage.ctor(Stream s, FileMode mode, FileAccess access, Boolean streaming)
Package.Open(Stream stream, FileMode packageMode, FileAccess packageAccess, Boolean streaming)
Tests.Cannot_Create_Package_On_Unseekable_Stream() line 22
```